include/boost/te.hpp

```diff
@@ -196,44 +428,64 @@ class static_vtable {
 
 namespace detail {
 struct poly_base {
-  detail::void_ptr ptr{};
-  detail::void_ptr::ptr_t *vptr{};
+  void** vptr = nullptr;
+  virtual void* ptr() const noexcept = 0;
 };
 }  // namespace detail
 
-template <class I, class TStorage = dynamic_storage,
-          class TVtable = static_vtable>
+template <
+  class I,
+  class TStorage = dynamic_storage,
+  class TVtable = static_vtable
+>
 class poly : detail::poly_base,
              public std::conditional_t<detail::is_complete<I>{}, I,
                                        detail::type_list<I> > {
  public:
-  template <class T,
-            class = std::enable_if_t<
-                not std::is_convertible<std::decay_t<T>, poly>{}> >
-  constexpr poly(T &&t) noexcept
+  template <
+    class T,
+    class T_ = std::decay_t<T>,
+    std::enable_if_t<!std::is_same_v<T_, poly>, bool> = true
+  >
+  constexpr poly(T &&t) noexcept(std::is_nothrow_constructible_v<T_,T&&>)
```

Comment:
`explicit` here?

Reply:
Yeah that won't compile for reasons

Reply:
The standard even says that an explicit templated constructor doesn't make sense

Reply:
So don't know how to make Codacy happy

Reply:
Explicit isn't used in Master branch either

Reply:
[This](https://en.cppreference.com/w/cpp/language/converting_constructor) page seems to suggest we don't want the 'explicit' specifier otherwise it restricts usage of the library, namely things like copy-initialization. @krzysztof-jusiak do you mind giving your opinion?

Reply:
I see, well, it is always possible to make the linter happy by adding `// cppcheck-suppress noExplicitConstructor` at line 449 (see man cppcheck). The invocation of cppcheck by Codacy suggests that inline suppressing instructions are supported: https://github.com/codacy/codacy-cppcheck/blob/67cc5350777af036c0a52455ae9073dfddc0918f/src/main/scala/codacy/cppcheck/CPPCheck.scala#L66

Reply:
Great, let's do that

Reply:
Don't think that worked

Reply:
I think this way the tool tries to apply the comment to `>`. Instead, perhaps either place the comment on the free line:
```
...
std::enable_if_t<!std::is_same_v<T_, poly>, bool> = true>
// cppcheck-suppress noExplicitConstructor
constexpr poly(T &&t) noexcept(std::is_nothrow_constructible_v<T_,T&&>)
```

or on the same line as to-be-ignored aspect of constructor:

```
...
std::enable_if_t<!std::is_same_v<T_, poly>, bool> = true>
constexpr poly(T &&t) // cppcheck-suppress noExplicitConstructor
   noexcept(std::is_nothrow_constructible_v<T_,T&&>)
```

I think either one should work.

Reply:
@hgkjshegfskef Thank you, that worked.